### PR TITLE
Reference naming improvements

### PR DESCRIPTION
**Pull request by GiovanH**
If there is no name info for references, get ref name from link.
Strip some punctuation from automatic reference names

### CODE REVIEW

**Comment by GiovanH** [comment]
Demo and use cases:
image
image

Previous behavior:
image

**Comment by GiovanH** [comment]
As of dc9fa53 this also fixes another bug with organize references not properly selecting references

File: plugins/references.py [hunk 1]
```diff
@@ -69,7 +71,7 @@ def getMarkers(view, name=""):
             markers.extend(view.find_all(r"(?<=\[)(%s)(?=\])(?!\s*\]:)" % name, 0))
     regions = []
 
-    selector = marker_text_scope_name + ", " + marker_text_scope_name
+    selector = marker_text_scope_name + ", " + marker_link_reference_scope_name
```
Comment (deathaxe) [inline plugins/references.py:74]: The selector `selector = marker_text_scope_name + ", " + marker_text_scope_name` contains duplicate scope names by accident. It should rather look like `selector = marker_ref_scope_name + ", " + marker_text_scope_name` to match the original behavior of the current and recent releases.

The `marker_ref_scope_name` would already match the name within brackets, only. I guess we therefore don't need to introduce another marker scope name?
    


My selector was wrong. It needs to look like

**Review by deathaxe — COMMENTED** [review]
Thanks for contributing.

**Comment by deathaxe** [comment]
I've resolved conflicts with changelog locally and squash merged your PR. Unfortunatelly GH doesn't recognize it as merged.

**Comment by GiovanH** [comment]
Oops, sorry for missing that. Thanks!

**Pull request closed without merging**